src/model/mod.rs

```diff
@@ -458,6 +459,9 @@ pub struct PairPileup<'a, A, B, P> where
     prior_model: &'a P,
     case_sample_model: likelihood::LatentVariableModel,
     control_sample_model: likelihood::LatentVariableModel,
+    // these two caches are useful for PairModels where case and control are independent of each other and Events are just combinations of across the two axes, as e.g. in the single-cell-bulk model
+    case_likelihood_cache: &'a mut BTreeMap<AlleleFreq, LogProb>,
+    control_likelihood_cache: &'a mut BTreeMap<AlleleFreq, LogProb>,
```

Comment:
These have to be owned by PairPileup. I.e., don't use a reference here, just `BTreeMap<AlleleFreq, LogProb>`. Have a look at this again: https://doc.rust-lang.org/book/second-edition/ch04-01-what-is-ownership.html#ownership-rules
PairPileup must own the caches.

Reply:
I think the sized issue might go away once this is fixed.